x-pack/plugins/alerting/server/rules_client/lib/siem_legacy_actions/transform_legacy_actions.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 } from 'uuid';
+import { isEmpty } from 'lodash/fp';
+import type { SavedObjectReference } from '@kbn/core/server';
+import { RawRuleAction } from '../../../types';
+import { transformToNotifyWhen } from './transform_to_notify_when';
+import { LegacyIRuleActionsAttributes } from './types';
+
+/**
+ * @deprecated Once we are confident all rules relying on side-car actions SO's have been migrated to SO references we should remove this function
+ * transforms siem legacy actions {@link LegacyIRuleActionsAttributes} objects into {@link RawRuleAction}
+ * @param legacyActionsAttr
+ * @param references
+ * @returns array of RawRuleAction
+ */
+export const transformFromLegacyActions = (
+  legacyActionsAttr: LegacyIRuleActionsAttributes,
+  references: SavedObjectReference[]
+): RawRuleAction[] => {
+  const actionReference = references.reduce<Record<string, SavedObjectReference>>(
+    (acc, reference) => {
+      acc[reference.name] = reference;
+      return acc;
+    },
+    {}
+  );
+
+  if (isEmpty(actionReference)) {
```

Comment:
do we need to throw an error, maybe we can just ignore it and return empty array?

Reply:
added try/catch in `formatLegacyActions` and `migrateLegacyActions` on top level. So any thrown errors will be caught and logged